modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/PublicationUserNotificationHandlerTest.java

```diff
@@ -435,14 +438,31 @@ private ServiceContext _getServiceContext() throws Exception {
 		ServiceContext serviceContext =
 			ServiceContextTestUtil.getServiceContext(_group.getGroupId());
 
-		serviceContext.setRequest(new MockHttpServletRequest());
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			_companyLocalService.fetchCompany(TestPropsValues.getCompanyId()));
+
+		themeDisplay.setRequest(mockHttpServletRequest);
+		themeDisplay.setSiteGroupId(_group.getGroupId());
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		serviceContext.setRequest(mockHttpServletRequest);
```

Comment:
Why do we need to set the themeDisplay? We didn't need it to create the show conflicts link.

Reply:
When we do not set it the link is created as null  [here](https://github.com/liferay-publications/liferay-portal/blob/bf93fc3ce4fe25e2677d5521def7772296f160b8/modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/test/PublicationUserNotificationHandlerTest.java#L248-L253), because of some [properties that are not initialized](https://github.com/liferay-publications/liferay-portal/blob/97039be0c54337fe719879d08bd5f83cb0108456/portal-kernel/src/com/liferay/portal/kernel/portlet/RequestBackedPortletURLFactoryUtil.java#L123-L132). The test just wasn't failing because both links were null so the assertion considered they the same

Reply:
Got it. You can leave it as is.